log/zap/field.go

```diff
@@ -59,3 +61,158 @@ func (t trace) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	enc.AddBool("sampled", j.IsSampled())
 	return nil
 }
+
+type spanContext struct {
+	spanContext jaeger.SpanContext
+}
```

Comment:
I though an alias would work just as well `type spanContext jaeger.SpanContext`

Reply:
I'm not sure that you can access methods when using type aliases. Let me verify

Reply:
So - if we were to use `type spanContext jaeger.SpanContext`, we'd have to cast the receiver to `jaeger.SpanContext` to access methods defined on `jaeger.SpanContext`. 

This is because `type T1 T2` defines a new type.

Reply:
yes, but you're already doing `ctx := s.spanContext`, instead can do `ctx := jaeger.SpanContext(s)`